src/LocalFetcher.ts

```diff
@@ -22,10 +22,17 @@ export default class LocalFetcher extends Fetcher {
     return path.resolve(this.containerXmlPath, '../../', relativeOpfPath);
   }
 
-  resolvePath(from: string, to: string): string {
+  resolvePath(from: string, to: string): string;
+  resolvePath(from: string, to: string | undefined): string | undefined;
+  resolvePath(from: string, to: unknown): unknown {
```

Comment:
Yes, the type checking works. Here's an example where the second param is a string: https://github.com/NYPL/ePub-to-webpub/blob/46b1bfae42b09f1038b58eea4e0a9c2667ddb3cb/src/convert/resourcesAndReadingOrder.ts#L97

I think this is how Typescript handles overloads. You want to declare all the possible variations of the method, without the actual implementation, and declare one concrete method but expect you to handle all different cases in that concrete implementation.

But just come to think about this a second time, I don't think we need the 'unknown' type for this. `string | undefined` is sufficient in our use case because that covers all the overloads. (I will update this)